lib/boutiques_scrnabox_ressource_manager.rb

```diff
@@ -0,0 +1,99 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2022
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+module BoutiquesScrnaboxRessourceManager
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  def update_mem(step_mem, mem)
```

Comment:
Don't do that, it makes the code way to complicated. Ther eis no need for a method that takes two params, compare them and returns the max.

Instead, in your loop below, just extract the value and make a quick compare/switch

```ruby
step_mem = suggested_resources_by_steps[step]["mem"]
mem = step_mem if step_mem > mem
```

Reply:
I see you need to find the max in other special steps for step 7 and 8 too.

Then just create an array of mems at the beginning

```ruby
mems = []
```

then push all the mems of the selected steps

```
mems << suggested_resources_by_steps[step]["mem"] # at each selected step
```

and at the end

mem = mems.max